### PR TITLE
Stm32mp1 i2c arduino

### PR DESCRIPTION
**Pull request by cpriouzeau**
Add support of I2C on arduino connector fro stm32mp157c-dk2.

Tested via samples/shields/x_nucleo_iks01a2.

Depends on patch from  #17035

### CODE REVIEW

**Review by arnopo — CHANGES_REQUESTED** [review]

File: boards/arm/stm32mp157c_dk2/Kconfig.defconfig [hunk 1]
```diff
@@ -26,8 +26,6 @@ endif # SERIAL
 
 if I2C
 
-#config I2C_STM32_V2
-#	default y
 config I2C_5
```
[the remark this answers is not in the document]
Comment (cpriouzeau) [reply boards/arm/stm32mp157c_dk2/Kconfig.defconfig]: @arnop2 Take into account on new push

File: soc/arm/st_stm32/stm32mp1/Kconfig.defconfig.series [hunk 2]
```diff
@@ -40,4 +40,11 @@ config GPIO_STM32_PORTK
 
 endif # GPIO_STM32
 
+if I2C_STM32
```
Comment (erwango) [inline soc/arm/st_stm32/stm32mp1/Kconfig.defconfig.series]: This change would better fit in previous commit.

**Review by erwango — CHANGES_REQUESTED** [review]

File: soc/arm/st_stm32/stm32mp1/Kconfig.defconfig.series (hunk 2 above)
Comment (cpriouzeau) [reply soc/arm/st_stm32/stm32mp1/Kconfig.defconfig.series]: Done

File: boards/arm/stm32mp157c_dk2/Kconfig.defconfig [hunk 3]
```diff
@@ -24,4 +24,13 @@ config UART_7
 
 endif # SERIAL
 
+if I2C
+
+config I2C_5
+	default y
+config I2C_5_IRQ_PRI
```
Comment (ydamigos) [inline boards/arm/stm32mp157c_dk2/Kconfig.defconfig]: We define IRQ priority in DT for STM32 I2C. Please remove it.

File: drivers/i2c/Kconfig.stm32 [hunk 4]
```diff
@@ -23,7 +23,7 @@ config I2C_STM32_V1
 
 config I2C_STM32_V2
 	bool
-	depends on SOC_SERIES_STM32F0X || SOC_SERIES_STM32F3X || SOC_SERIES_STM32F7X || SOC_SERIES_STM32L0X || SOC_SERIES_STM32L4X || SOC_SERIES_STM32WBX
+	depends on SOC_SERIES_STM32F0X || SOC_SERIES_STM32F3X || SOC_SERIES_STM32F7X || SOC_SERIES_STM32L0X || SOC_SERIES_STM32L4X || SOC_SERIES_STM32WBX || SOC_SERIES_STM32MP1X
```
Comment (ydamigos) [inline drivers/i2c/Kconfig.stm32]: You could also update help section.

**Review by ydamigos — CHANGES_REQUESTED** [review]

File: drivers/i2c/Kconfig.stm32 (hunk 4 above)
Comment (cpriouzeau) [reply drivers/i2c/Kconfig.stm32]: Done with next push

File: boards/arm/stm32mp157c_dk2/Kconfig.defconfig (hunk 3 above)
Comment (cpriouzeau) [reply boards/arm/stm32mp157c_dk2/Kconfig.defconfig]: Done with next push

**Comment by yboutreux** [comment]
@cpriouzeau FYI https://github.com/zephyrproject-rtos/zephyr/pull/17035 has been merged

**Review by arnopo — APPROVED** [review]

**Review by erwango — CHANGES_REQUESTED** [review]
Please split following commit in 2: board and soc:
https://github.com/zephyrproject-rtos/zephyr/pull/17044/commits/4fcdee5f0676150dc1bfc64ef51fd2023560ab82

Also, a rebase would be required.
Otherwise LGTM

**Review by erwango — APPROVED** [review]

**Comment by erwango** [comment]
Approved. `DNM` waiting https://github.com/zephyrproject-rtos/hal_stm32/pull/14 to be merged (and west commit updated)

**Review by ydamigos — APPROVED** [review]

**Comment by erwango** [comment]
@cpriouzeau, dependency has been merged, can you update west.yml using correct SHA1 for the module dependency and push again?

**Comment by arnop2** [comment]
As @cpriouzeau  is in vacation, i rebased it in a new pull request  #17807
This one can be closed

**Pull request closed without merging**